src/main/steps/applicant1/finalising-your-application/post.ts

```diff
@@ -19,7 +30,39 @@ export default class FinalisingYourApplicationPostController extends PostControl
     return super.save(req, formData, eventName);
   }
 
+  protected getNextUrl(req: AppRequest): string {
+    const hasApplicant2SwitchedToSoleFo =
+      req.session.isApplicant2 &&
+      req.originalUrl === APPLICANT_2 + FINALISING_YOUR_APPLICATION &&
+      req.session.userCase.finalOrderSwitchedToSole === YesOrNo.YES &&
+      req.session.userCase.state === State.FinalOrderRequested &&
```

Comment:
Need to be careful when the case is Welsh. Should be looking at Welsh previous state

Reply:
Could just leave this line out if thats gonna complicate this

Reply:
Removed the state check - the other logical tests should be sufficient.